sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -35,3 +37,57 @@ case class Size(child: Expression) extends UnaryExpression with ExpectsInputType
     nullSafeCodeGen(ctx, ev, c => s"${ev.primitive} = ($c).size();")
   }
 }
+
+case class ArrayContains(left: Expression, right: Expression) extends BinaryExpression {
+  override def dataType: DataType = BooleanType
+
+  override def checkInputDataTypes(): TypeCheckResult = {
+    if (!left.dataType.isInstanceOf[ArrayType]) {
+      TypeCheckResult.TypeCheckFailure(
+        s"type of first input must be an array, not ${left.dataType.simpleString}")
+    } else if (left.dataType.asInstanceOf[ArrayType].elementType != right.dataType) {
+      TypeCheckResult.TypeCheckFailure(
+        s"type of value must match array type " +
+          s"${left.dataType.asInstanceOf[ArrayType].elementType.simpleString}, not "+
+          s"${right.dataType.simpleString}")
+    } else {
+      TypeCheckResult.TypeCheckSuccess
+    }
+  }
+
+  override def eval(input: InternalRow): Boolean = {
```

Comment:
Would this still let it satisfy that it returns false if either left or right is null? I was under the impression it would return null using `nullSafeEval`